lib/Sema/TypeCheckCodeCompletion.cpp

```diff
@@ -527,8 +556,20 @@ static void filterSolutions(SolutionApplicationTarget &target,
   })->getFixedScore();
 
   llvm::erase_if(solutions, [&](const Solution &S) {
-    return S.getFixedScore().Data[SK_Fix] != 0 &&
-        S.getFixedScore() > minScore;
+    if (S.getFixedScore().Data[SK_Fix] > minScore.Data[SK_Fix]) {
+      return true;
+    } else if (S.getFixedScore().Data[SK_Fix] == minScore.Data[SK_Fix]) {
+      // We don't care how much source code was ignored after the code
+      // completion token, just whether if there was any.
+      if (S.getFixedScore().Data[SK_IgnoreAfterCompletionToken] > 0 &&
+          minScore.Data[SK_IgnoreAfterCompletionToken] == 0) {
```

Comment:
I am not really happy with this part yet. What I am really trying to achieve is to have a boolean flag in the score that indicates whether code after the code completion token was ignored – I don’t care how much.  At the moment we might end up with solutions that have a `SK_IgnoreAfterCompletionToken` score of >1 (sometimes also created by combining two partial solutions, where each partial solution ignored arguments). 

In the current implementation, we might filter out solutions with a `SK_IgnoreAfterCompletionToken` of 2 when comparing against a solution that has a `SK_IgnoreAfterCompletionToken` of 1 in the constraint system, even though we don’t really consider it worse.

Options that I have considered:
- Only increase score in `recordFix` if `SK_IgnoreAfterCompletionToken` is 0 — doesn’t work because we might combine two partial solutions that both have `SK_IgnoreAfterCompletionToken = 1`
- Hacking the `operator<` of `Score` to consider all `SK_IgnoreAfterCompletionToken > 0` equally bad — this seems very surprising to me
- Hack the `operator+` of `Score` such that `1 + 1 = 1` for `SK_IgnoreAfterCompletionToken` — this makes `Score` addition non-linear, not sure what effects it might have
- Leave it as-is for now because we don’t have any counter-examples that break yet